lib/internal/crypto/crypto.js

```diff
@@ -0,0 +1,525 @@
+
+    super(new binding.DiffieHellman(sizeOrKey, generator));
+  }
+
+  setPublicKey(key, encoding) {
+    encoding = encoding || exports.DEFAULT_ENCODING;
+    this._handle.setPublicKey(toBuf(key, encoding));
+    return this;
+  }
+
+  setPrivateKey(key, encoding) {
+    encoding = encoding || exports.DEFAULT_ENCODING;
+    this._handle.setPrivateKey(toBuf(key, encoding));
+    return this;
+  }
+
+  [inspect](depth, options) {
+    var obj = {
+      generator: this.getGenerator('hex'),
+      prime: this.getPrime('hex')
+    };
+    if (this._options)
+      obj.options = this._options;
+    return `DiffieHellman ${util.inspect(obj, options)}`;
+  }
+}
+
+class DiffieHellmanGroup extends DiffieHellmanBase {
+  constructor(name) {
+    super(new binding.DiffieHellmanGroup(name));
+    this[kAlgorithm] = name;
+  }
+
+  [inspect](depth, options) {
+    var obj = {
+      name: this[kAlgorithm],
+      generator: this.getGenerator('hex'),
+      prime: this.getPrime('hex')
+    };
+    if (this._options)
+      obj.options = this._options;
+    return `DiffieHellmanGroup ${util.inspect(obj, options)}`;
+  }
+}
+
+class ECDH {
+  constructor(curve) {
+    this._handle = new binding.ECDH(curve);
+    this[kAlgorithm] = curve;
+  }
+
+  generateKeys(encoding, format) {
+    this._handle.generateKeys();
+    return this.getPublicKey(encoding, format);
+  }
+
+  getPublicKey(encoding, format) {
+    var f = constants.POINT_CONVERSION_UNCOMPRESSED;
+    if (format !== undefined) {
+      switch (format) {
+        case 'compressed':
+          f = constants.POINT_CONVERSION_COMPRESSED;
+          break;
+        case 'hybrid':
+          f = constants.POINT_CONVERSION_HYBRID;
+          break;
+        case 'uncompressed':
+          f = constants.POINT_CONVERSION_UNCOMPRESSED;
+          break;
+        default:
+          throw new TypeError(`Bad format: ${format}`);
+      }
+    }
+    var key = this._handle.getPublicKey(f);
+    encoding = encoding || exports.DEFAULT_ENCODING;
+    if (encoding && encoding !== 'buffer')
+      key = key.toString(encoding);
+    return key;
+  }
+
+  [inspect](depth, options) {
+    var obj = {
+      curve: this[kAlgorithm]
+    };
+    if (this._options)
+      obj.options = this._options;
+    return `ECDH ${util.inspect(obj, options)}`;
+  }
+}
+ECDH.prototype.computeSecret = DiffieHellman.prototype.computeSecret;
+ECDH.prototype.setPrivateKey = DiffieHellman.prototype.setPrivateKey;
+ECDH.prototype.setPublicKey = DiffieHellman.prototype.setPublicKey;
+ECDH.prototype.getPrivateKey = DiffieHellman.prototype.getPrivateKey;
+
+class Certificate {
+  verifySpkac(object) {
+    return binding.certVerifySpkac(object);
+  }
+
+  exportPublicKey(object, encoding) {
+    return binding.certExportPublicKey(toBuf(object, encoding));
+  }
+
+  exportChallenge(object, encoding) {
+    return binding.certExportChallenge(toBuf(object, encoding));
+  }
+}
+
+exports.Hash = Hash;
```

Comment:
Can we convert these to use the `module.exports = { ... }` format instead? The reason is that at least `exports.DEFAULT_ENCODING` is accessed quite frequently in the crypto code here and having a faster property lookup would be beneficial.